core/src/main/scala/org/apache/spark/sql/SnappySession.scala

```diff
@@ -1402,6 +1434,36 @@ class SnappySession(_sc: SparkContext) extends SparkSession(_sc) {
     }
   }
 
+  private[sql] def alterTableToggleRLS(tableIdent: QualifiedTableName, enableRls: Boolean): Unit = {
+    val plan = try {
+      sessionCatalog.lookupRelation(tableIdent)
+    } catch {
+      case tnfe: TableNotFoundException => throw tnfe
+    }
+
+    if(sessionCatalog.isTemporaryTable(tableIdent)) {
+      throw new AnalysisException("alter table not supported for temp tables")
+    }
+
+    SnappyContext.getClusterMode(sc) match {
+      case ThinClientConnectorMode(_, _) =>
+        throw new AnalysisException("alter table enable/disable Row Level Security not supported " +
+            "for smart connector mode")
+      case _ =>
+    }
+
+    plan match {
+      case LogicalRelation(rls: RowLevelSecurityRelation, _, _) =>
+        sessionCatalog.invalidateTable(tableIdent)
+        rls.enableOrDisableRowLevelSecurity(tableIdent, enableRls)
+        // TODO : Asif : Is it needed what does this do?
+        SnappyStoreHiveCatalog.registerRelationDestroy()
```

Comment:
This will force refetch of meta-data by smart connector (bucket info etc) among other things. Perhaps can be avoided but does not harm and good to have.